WalletWasabi.Fluent/ApplicationStateManager.cs

```diff
@@ -135,20 +136,30 @@ private void CreateAndShowMainWindow()
 			.TakeWhile(_ => !_isShuttingDown) // Prevents stack overflow.
 			.Subscribe(tup =>
 			{
-				// _hideRequest flag is used to distinguish what is the user's intent.
-				// It is only true when the request comes from the Tray.
-				if ((Services.UiConfig.HideOnClose || _hideRequest) && App.EnableFeatureHide)
+				// Check if Wasabi was forcefully terminated from terminal with Ctrl-C.
+				if (!Services.TerminateService.ForcefulTerminationRequestedTask.IsCompletedSuccessfully)
 				{
-					_hideRequest = false; // request processed, set it back to the default.
-					return;
-				}
+					// _hideRequest flag is used to distinguish what is the user's intent.
+					// It is only true when the request comes from the Tray.
+					if ((Services.UiConfig.HideOnClose || _hideRequest) && App.EnableFeatureHide)
+					{
+						_hideRequest = false; // request processed, set it back to the default.
+						return;
+					}
 
-				var (e, preventShutdown) = tup;
+					var (e, preventShutdown) = tup;
 
-				_isShuttingDown = !preventShutdown;
-				e.Cancel = preventShutdown;
+					_isShuttingDown = !preventShutdown;
+					e.Cancel = preventShutdown;
 
-				_stateMachine.Fire(preventShutdown ? Trigger.ShutdownPrevented : Trigger.ShutdownRequested);
+					_stateMachine.Fire(preventShutdown ? Trigger.ShutdownPrevented : Trigger.ShutdownRequested);
+				}
+				else
+				{
+					_isShuttingDown = true;
+					tup.EventArgs.Cancel = false;
+					_stateMachine.Fire(Trigger.ShutdownRequested);
+				}
```

Comment:
Invert the if with a return and change your comment for `// Bypass shutdown prevention if Wasabi was forcefully terminated from terminal with Ctrl-C.`